### DISCUSSION TITLE
Timestamp issues, why timestamp in RTSP is lesser then the one embedded on a frame?

### DISCUSSION THREAD

**Discussion by TrueWodzu** [bluenviron/mediamtx#3910] [Questions]
### Question

Hello,

I've created simple video on which I am embedding a timestamp and then I am sending that video to MediaMTX, here is a basic command for encoding each frame:

`"appsrc ! videoconvert ! x264enc speed-preset=ultrafast ! rtspclientsink location=rtsp://127.0.0.1:8554/mystream"`

Then, I am retrieving each frame via RTSP stream and I am embedding a second timestamp on the image. The second one is a timestamp obtained from RTSP, the one that is added to each packet. The problem is that the second timestamp is incorrect, quite often it is earlier then the one that I've embedded, for example:

![20241024T145049+0200](https://github.com/user-attachments/assets/51bb209d-0444-400e-b047-4d332cf8c38e)

The white timestamnp is the one that I've embedded into the frame before sending it to MediaMTX via GStreamer, the blue one is obtained from RTSP. This blue one should never be less than the white one, but it happens quite often. Also the difference between two timestamps is fluctuating very much.

Nothing like this is happening when i am receiving a stream from an actual camera. Then, the difference is always constant and the blue one is always bigger than the white one.

Encoding application and MediaMTX are on the same machine, but MediaMTX is working in docker container. Any help would be much appreciated.

**Comment by aler9** [comment]
Hello, by default MediaMTX replaces the RTSP absolute timestamp (and by that i mean the absolute timestamp in RTCP reports) with the timestamp corresponding to the instant in which a packet has been received by the server, which is subjected to fluctuations. This happens because many cameras and protocols do not provide absolute timestamps, so the only way to provide them is to regenerate them from scratch.

You can route original absolute timestamps instead of replacing them by toggling `useAbsoluteTimestamp` in the configuration:

```yml
pathDefaults:
  # Use original absolute timestamps of RTSP and WebRTC frames.
  # In RTSP and WebRTC, absolute timestamps are transported through RTCP reports.
  useAbsoluteTimestamp: false
```
